### PR TITLE
Issue #915 fix clip_repeat_stress when time_start or time_end is None

### LINKED ISSUE #915: imod.mf6.Modflow6Simulation.clip_box fails for undefined time_start and time_end?
Running this:

```python
simulation = simulation.clip_box(layer_min=1, layer_max=10)
```

Gives me:

```
--> 349 keep = (keys >= time_start) & (keys <= time_end)
    350 new_keys = keys[keep]
    351 new_values = values[keep]

TypeError: '>=' not supported between instances of 'int' and 'NoneType'
```

It errors in _clip_repeat_stress.

Checking for None fixes it:

```python
        if time_start is None or time_end is None:
            keep = slice(None, None)
        else:
            keep = (keys >= time_start) & (keys <= time_end)
```

Another option is just skipping the work of _clip_repeat_stress if time_start and time_end are None.

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #915

# Description
- Adds extra tests for ``clip_repeat_stress`` where ``time_start`` or ``time_end`` is None
- In ``clip_repeat_stress``, make clipping work with ``time_start`` or ``time_end`` is None
- In ``clip_repeat_stress``, prepend repeated values before lookup keys, to ensure these values are accounted for in the clipping and subsequent promoting of keys.

# Checklist
- [x] Links to correct issue
- [x] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/tests/test_common/test_utilities/test_clip.py [hunk 1]
```diff
@@ -402,6 +402,12 @@ def test_clip_repeat_stress__all_repeats(dataset):
     values = repeat_stress.loc[:, 1]
     actual = dataset.drop_vars("time").isel(time=indexer)
     assert actual["time"].size == 12
+    # Assert
```
Comment (Manangka) [inline imod/tests/test_common/test_utilities/test_clip.py:405]: Could you add a comment to this test what you are testing?

I don't get why you would expect that it repeats 12 times
I also don't get what the array_equal asserts test. They values are different in the tests below so it seems relevant.

Comment (JoerivanEngelen) [reply imod/tests/test_common/test_utilities/test_clip.py:414]: I tried clarifying the tests more
